### ISSUE TITLE
Move setup into `core` module

### ISSUE THREAD

**Issue by VonHeikemen** [AstroNvim/AstroNvim#998]
### Is your feature related to a problem?

Not really.

### Describe the new feature

Move the setup from `init.lua` to a module in `lua/core`.

I'm not asking you to delete `init.lua` or anything like that. I just want to call that code using `require`.

### Additional context

I would like to use astronvim along side my regular neovim configuration. If I could just require the code in `init.lua` it could make it easier to call astronvim like a plugin.

So I want to clone astronvim in a different directory away from `~/config/nvim`. And then use `nvim -u` to point neovim to a custom entrypoint. In this new entrypoint I can make sure astronvim and my own config don't have any conflict.

Now, I already have a workaround for this using the `source` command, so I'm not looking for a code snippet to make it work. I just want it to be slightly easier.

**Comment by mehalter** [comment]
I'm not sure I follow how this would change things, We would need to keep the `init.lua` file intact for usability purposes and you wouldn't be able to use it and modify it without breaking the update path for AstroNvim. You can still require each of the individual parts of core, so I don't think I see much of an added benefit to this. Could you please share exactly how you plan to use it and what sort of directory structure you are looking at? You can also clone it to a different folder and in your `~/.config/nvim/init.lua` file add the new location to your runtimepath and it will be picked up without messing up your `~/.config/nvim` folder.

**Comment by VonHeikemen** [comment]
So like moving the code from `init.lua` to a lua module would allow more advance user to load astronvim's config easily. And you could use the same module in your `init.lua`, no need to duplicate code. Is not big deal really, it'll just be nice.

I shared my workaround in the docs repository: https://github.com/AstroNvim/astronvim.github.io/issues/12#issuecomment-1257176191

**Comment by mehalter** [comment]
I don't think this is something we need to do and I don't know how this will change loading of AstroNvim for advanced users since wherever you put AstroNvim you will still need to source a file no matter what, it doesn't necessarily mean that you can load things more easily so I am going to close this. If someone can explain in detail how they would use this and why you cannot do the same method with the current set up I can reopen this and take a look
